guacamole-ext/src/main/java/org/apache/guacamole/environment/Environment.java

```diff
@@ -118,20 +124,103 @@ public <Type> Type getProperty(GuacamoleProperty<Type> property)
      * property in guacamole.properties, if any. If no value is found, the
      * provided default value is returned.
      *
-     * @param <Type> The type that the given property is parsed into.
-     * @param property The property to read from guacamole.properties.
-     * @param defaultValue The value to return if no value was given in
-     *                     guacamole.properties.
-     * @return The parsed value of the property as read from
-     *         guacamole.properties, or the provided default value if no value
-     *         was found.
-     * @throws GuacamoleException If an error occurs while parsing the value
-     *                            for the given property in
-     *                            guacamole.properties.
+     * @param <Type>
+     *     The type that the given property is parsed into.
+     * 
+     * @param property
+     *     The property to read from guacamole.properties.
+     * 
+     * @param defaultValue
+     *     The value to return if no value was given in guacamole.properties.
+     * 
+     * @return
+     *     The parsed value of the property as read from guacamole.properties,
+     *     or the provided default value if no value was found.
+     * 
+     * @throws GuacamoleException
+     *     If an error occurs while parsing the value for the given property in
+     *     guacamole.properties.
      */
     public <Type> Type getProperty(GuacamoleProperty<Type> property,
             Type defaultValue) throws GuacamoleException;
 
+    /**
+     * Given a GuacamoleProperty, parses and returns a sorted Collection of the
+     * value set for that property in guacamole.properties, if any.
```

Comment:
As it's up to the property implementation to support being read as a `Collection`, I think we should document here and in other `getPropertyCollection()` flavors that not all properties support being read as multiple values and that it's up to the property implementation to define how multiple values are parsed, if at all.